cpp/src/strings/convert/convert_ipv4.cu

```diff
@@ -0,0 +1,230 @@
+ * This means "128-34-56-709" will parse successfully.
+ */
+struct ipv4_to_integers_fn
+{
+    column_device_view const d_strings;
+
+    __device__ int64_t operator()( size_type idx )
+    {
+        if( d_strings.is_null(idx) )
+            return 0;
+        string_view d_str = d_strings.element<string_view>(idx);
+        int32_t ipvals[4] = {0}; // IPV4 format: xxx.xxx.xxx.xxx
+        int32_t ipv_idx = 0, factor = 1;
+        const char* in_ptr = d_str.data();
+        const char* end = in_ptr + d_str.size_bytes();
+        while( (in_ptr < end) && (ipv_idx < 4))
+        {
+            char ch = *in_ptr++;
+            if( ch < '0' || ch > '9' )
+            {
+                ++ipv_idx;
+                factor = 1;
+            }
+            else
+            {
+                ipvals[ipv_idx] = (ipvals[ipv_idx]*factor) + static_cast<int32_t>(ch-'0');
+                factor = 10;
+            }
+        }
+        return static_cast<int64_t>(ipvals[0] << 24) + (ipvals[1] << 16) + (ipvals[2] << 8) + ipvals[3];
+    }
+};
+
+}
+
+// Convert strings column of IPv4 addresses to integers column
+std::unique_ptr<column> ipv4_to_integers( strings_column_view const& strings,
+                                          rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                          cudaStream_t stream = 0)
+{
+    size_type strings_count = strings.size();
+    if( strings_count == 0 )
+        return make_numeric_column( data_type{INT64}, 0 );
+
+    auto strings_column = column_device_view::create(strings.parent(), stream);
+    auto d_strings = *strings_column;
+    // create output column copying the strings' null-mask
+    auto results = make_numeric_column( data_type{INT64}, strings_count,
+        copy_bitmask( strings.parent(), stream, mr), strings.null_count(), stream, mr);
+    auto d_results = results->mutable_view().data<int64_t>();
+    // fill output column with ipv4 integers
+    thrust::transform( rmm::exec_policy(stream)->on(stream),
+        thrust::make_counting_iterator<size_type>(0),
+        thrust::make_counting_iterator<size_type>(strings_count),
+        d_results, ipv4_to_integers_fn{d_strings});
+    // done
+    results->set_null_count(strings.null_count());
+    return results;
+}
+
+} // namespace detail
+
+// external API
+std::unique_ptr<column> ipv4_to_integers( strings_column_view const& strings,
+                                         rmm::mr::device_memory_resource* mr)
+{
+    return detail::ipv4_to_integers(strings, mr);
+}
+
+namespace detail
+{
+namespace
+{
+
+/**
+ * @brief Converts integers into IPv4 addresses.
+ *
+ * Each integer is divided into 8-bit sub-integers.
+ * The sub-integers are converted into 1-3 character digits.
+ * These are placed appropriately between '.' character.
+ */
+struct integers_to_ipv4_fn
+{
+    column_device_view const d_column;
+    int32_t const* d_offsets;
+    char* d_chars;
```

Comment:
Similar comment to the `regex ops` review, would prefer to avoid raw pointers here in favor of `thrust` or `std` data structures.

Reply:
We are accessing disjoint and variable sized sections of device memory.
This is executed per device thread and the index is used to address the appropriate string as well as access the offset value to point to that varying memory section for the output.